contracts/Remittance.sol

```diff
@@ -15,52 +15,53 @@ contract Remittance is Stoppable {
+    function hashOTP(address exchange, bytes32 password) public view returns(bytes32 hash) {
         require(exchange != address(0) && password != 0, "Both exchange and password must be set");
         hash = keccak256(abi.encodePacked(address(this), exchange, password));
     }
     
-    function newTransaction(address exchange, bytes32 password) _onlyIfRunning public payable returns(bool success) {
-        require(exchange != address(0) && password != 0 && msg.value > 0, "An error occured. Ensure that both exchange and password are set and that your transaction value is above 0");
-        
-        bytes32 hash = hashOTP(exchange, password);
-        
-        require(transactions[hash].emitter == address(0), "Password already used");
+    function newTransaction(address exchange, bytes32 hashedOTP) _onlyIfRunning public payable returns(bool success) {
+        require(exchange != address(0) && hashedOTP != 0 && msg.value > 0, "An error occured. Ensure that both exchange and the secret are set and that your transaction value is above 0");
+
+        require(transactions[hashedOTP].emitter == address(0), "Password already used");
         
         uint amount = msg.value;
         if (getOwner() != msg.sender) {
-            fees = fees.add(TX_FEES);
+            fees[getOwner()] = fees[getOwner()].add(TX_FEES);
             amount = amount.sub(TX_FEES);
             emit LogTakeFees(msg.sender, TX_FEES);
         }
         
         emit LogNewTransaction(msg.sender, exchange, amount);
-        transactions[hash] = Transaction({time: now.add(CANCELLATION_DELAY), emitter: msg.sender, amount: amount});
+        transactions[hashedOTP] = Order({
+            time: now.add(CANCELLATION_DELAY),
+            emitter: msg.sender,
+            amount: amount
+        });
         
         return true;
     }
     
-    function cancelTransaction(address exchange, bytes32 password) public returns(bool success) {
-        require(exchange != address(0) && password != 0, "Both exchange and password must be set");
+    function cancelTransaction(address exchange, bytes32 hashedOTP) public returns(bool success) {
+        require(hashedOTP != 0, "The secret is not set");
         
-        bytes32 hash = hashOTP(exchange, password);
-        Transaction memory tx = transactions[hash];
+        Order memory tx = transactions[hashedOTP];
         
         require(tx.amount > 0, "Neither the transaction does not exist or the password is wrong");
         require(now >= tx.time, "You must wait 5 days before cancelling the transaction");
         require(tx.emitter == msg.sender, "The call must be initiated by the wanted exchange");
         
-        transactions[hash].amount = 0;
-        transactions[hash].time = 0;
-        transactions[hash].emitter = address(0);
+        transactions[hashedOTP].amount = 0;
+        transactions[hashedOTP].time = 0;
+        transactions[hashedOTP].emitter = address(0);
```

Comment:
You could say `delete transactions[hashedOTP]`.

Are you intentionally allowing reuse of previously used passwords?

Reply:
I forgot this utility ;)